clientlibrary/config/config_test.go

```diff
@@ -32,13 +33,36 @@ func TestConfig(t *testing.T) {
 		WithInitialPositionInStream(TRIM_HORIZON).
 		WithIdleTimeBetweenReadsInMillis(20).
 		WithCallProcessRecordsEvenForEmptyRecordList(true).
-		WithTaskBackoffTimeMillis(10)
+		WithTaskBackoffTimeMillis(10).
+		WithEnhancedFanOutConsumer("fan-out-consumer")
 
 	assert.Equal(t, "appName", kclConfig.ApplicationName)
 	assert.Equal(t, 500, kclConfig.FailoverTimeMillis)
 	assert.Equal(t, 10, kclConfig.TaskBackoffTimeMillis)
+	assert.True(t, kclConfig.EnableEnhancedFanOutConsumer)
+	assert.Equal(t, "fan-out-consumer", kclConfig.EnhancedFanOutConsumerName)
 
 	contextLogger := kclConfig.Logger.WithFields(logger.Fields{"key1": "value1"})
 	contextLogger.Debugf("Starting with default logger")
 	contextLogger.Infof("Default logger is awesome")
 }
+
+func TestEmptyEnhancedFanOutConsumerName(t *testing.T) {
+	assert.PanicsWithValue(t, "Non-empty value expected for EnhancedFanOutConsumerName, actual: ", func() {
```

Comment:
From https://github.com/awslabs/amazon-kinesis-client/blob/cce05bf4fd03eed007bf6c51203f3cccc189c92a/amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutConfig.java#L51, this is not necessarily a failure condition.